### PR TITLE
Move contentpal interfaces to separete module. Implements #83

### PR DESCRIPTION
**Pull request by dmfs**
This allows to create a testing module without circular dependencies.

### CODE REVIEW

File: contentpal/src/main/java/org/dmfs/android/contentpal/tools/ClosableEmptyIterator.java [hunk 1]
```diff
@@ -29,7 +30,7 @@
  *
  * @author Marten Gajda
  */
-public final class ClosableEmptyIterator<T> extends AbstractBaseIterator<T> implements ClosableIterator<T>
+public final class ClosableEmptyIterator<T> extends AbstractBaseIterator<T> implements Iterator<T>, ClosableIterator<T>
```
Comment (lemonboston) [inline contentpal/src/main/java/org/dmfs/android/contentpal/tools/ClosableEmptyIterator.java]: Why are these explicit `Iterator<T>` declarations needed beside extending `CloseableIterator` here and in `FakeCloseable`?

Comment (dmfs) [reply contentpal/src/main/java/org/dmfs/android/contentpal/tools/ClosableEmptyIterator.java]: It isn't I guess Intellij added it when I deleted the `ClosableIterator` interface. I'll remove it.

Comment (dmfs) [reply contentpal/src/main/java/org/dmfs/android/contentpal/tools/ClosableEmptyIterator.java]: Fixed. @lemonboston please check again.

**Review by lemonboston — APPROVED** [review]
Looks good.

**Pull request merged**